.github/workflows/build_wheels.yml

```diff
@@ -0,0 +1,151 @@
+# a change to 'MCEq' directory
+# see https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request
+
+on:
+  workflow_dispatch:
+    # inputs:
+    #   artifactNameSDist:
+    #     description: 'Name of artifact for SDist'
+    #     required: true
+    #     default: 'SDist'
+    #   artifactNameWheels:
+    #     description: 'Prefix for artifacts for wheels'
+    #     required: true
+    #     default: 'wheels'
+  push:
+    branches:
+      - master
+    paths:
+      - 'MCEq/**'
+      - '!MCEq/data/README.md'
+    tags:        
+      - 'release**' 
+      # push tag release** (release_..., release/...) initiates upload to PyPI
+      # see also https://stackoverflow.com/a/71076957
+  pull_request:
+    types:
+      - opened
+    branches:    
+      - master
+    paths:
+      - 'MCEq/**'
+      - '!MCEq/data/README.md'
+
+jobs:
+    make_sdist:
+      name: Make SDist
+      runs-on: ubuntu-latest
+      steps:
+        - uses: actions/checkout@v3
+          # with:
+          # # number of commits to fetch, default: "1". "0" means all history
+          #   fetch-depth: 0
+
+        - name: Build SDist
+          run: pipx run build --sdist
+
+        - uses: actions/upload-artifact@v3
+          with:
+          # name: ${{ github.event.inputs.artifactNameSDist }}
+          # If "name" is not provided, "artifact" will be used as the default name
+            path: dist/*.tar.gz
+            
+    build_wheels:
+      name: Wheel on ${{ matrix.os }}
+      runs-on: ${{ matrix.os }}
+      strategy:
+        fail-fast: false
+        matrix:
+          os: [ubuntu-20.04, windows-2019, macos-10.15]
+
+      steps:
+      - uses: actions/checkout@v3
+        
+        # Building non-native architectures using emulation
+        # https://cibuildwheel.readthedocs.io/en/stable/faq/
+      - name: Set up QEMU
+        if: runner.os == 'Linux'
+        uses: docker/setup-qemu-action@v1
+        with:
+          platforms: all
+
+
+         
+      - name: Build wheels
+        uses: pypa/cibuildwheel@v2.5.0
+        env:
+          # https://cibuildwheel.readthedocs.io/en/stable/options/#examples_1
+          # Skip 32-bit builds
+          # Disable building PyPy wheels on all platforms
+          # Disable building musllinux wheels on all platforms
+          CIBW_SKIP: "*-win32 *linux_i686 pp* *musllinux*"
+          # Build `universal2` wheels on an Intel runner.
+          CIBW_ARCHS_MACOS: x86_64 universal2
+          # On an Linux Intel runner with qemu installed, build Intel and ARM wheels
+          CIBW_ARCHS_LINUX: "auto aarch64"
+          # Skip trying to test arm64 builds on Intel Macs
+          # Will avoid testing on emulated architectures
+          CIBW_TEST_SKIP: "*-macosx_arm64 *-macosx_universal2:arm64 *-*linux_{aarch64,ppc64le,s390x}"
+          # Run the package tests using 'pytest'
+          CIBW_TEST_REQUIRES: pytest
+          CIBW_TEST_COMMAND: pytest {package}/MCEq/tests
```

Comment:
I’m not sure what’s happening here.

What should happen is:

1. Build wheel
2. Install wheel via pip incl. the dependencies
3. run tests on the installed version via python3 -m pytest —pyargs MCEq (from outside the source directory)

Maybe this is what’s happening here?

Reply:
What I mean is that in the old ci scripts in `.ci/build_wheels.sh`, we had:
```
# Install packages and test
for PYBIN in "${pys[@]}"; do
    "${PYBIN}/python" -m pip install $package_name --no-index -f /io/wheelhouse
    "${PYBIN}/pip" install --prefer-binary -r /io/$test_requirements_file
    if [ -d "/io/tests" ]; then
        "${PYBIN}/pytest" /io/tests
    else
        if [ `uname -m` == "aarch64" ]; then
	   # Skipping test_mceq.py as the job gets timeout on aarch64
           cd /io/MCEq/tests
           "${PYBIN}/pytest" test_densities.py test_msis.py
        else
           "${PYBIN}/pytest" --pyargs $package_name
        fi
    fi
done
```
The freshly built wheels were installed by pip. The first if is not triggered for MCEq because there was no need to build binary tests. Then the code goes on all except aarch64 to the second else without entering the `MCEq/tests` directory. For aarch only a smaller subset of tests has been run. This method also tests the wheel's installation by pip including the requirements. Here the requirements are just pytest, so it seems to work.

Reply:
OK, solved. Thx